src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -163,162 +173,598 @@ public long GetNumberOfCountersToProcess(DocumentsOperationContext context, stri
+                            data = data.Clone(context);
+                            data.TryGet(DbIds, out BlittableJsonReaderArray dbIds);
+                            data.TryGet(Values, out BlittableJsonReaderObject localCounters);
+
+                            var localDbIdsList = DbIdsToList(dbIds);
+
+                            sourceData.TryGet(DbIds, out BlittableJsonReaderArray sourceDbIds);
+
+
+                            var prop = new BlittableJsonReaderObject.PropertyDetails();
+                            for (var i = 0; i< sourceCounters.Count; i++)
+                            {
+                                sourceCounters.GetPropertyByIndex(i, ref prop);
+
+                                var counterName = prop.Name;
+                                var modified = false;
+                                var changeType = CounterChangeTypes.Put;
+                                long value = 0;
+
+                                LazyStringValue deletedLocalCounter = null;
+                                BlittableJsonReaderObject.RawBlob loaclCounterValues = null;
+
+                                if (localCounters.TryGet(counterName, out object existingCounter))
+                                {
+                                    if (existingCounter is LazyStringValue lsv)
+                                    {
+                                        deletedLocalCounter = lsv;
+                                    }
+                                    else
+                                    {
+                                        loaclCounterValues = existingCounter as BlittableJsonReaderObject.RawBlob;
+                                    }
+                                }
+
+
+                                if (prop.Value is LazyStringValue deletedSourceCounter)
+                                {
+                                    if (deletedLocalCounter != null)
+                                    {
+                                        // delete + delete => merger change vectors
+
+                                        var mergedCv = ChangeVectorUtils.MergeVectors(deletedLocalCounter, deletedSourceCounter);
+                                        // todo add local dbId+etag to deleted change vector
+
+                                        localCounters.Modifications = localCounters.Modifications ?? new DynamicJsonValue(localCounters);
+                                        localCounters.Modifications[counterName] = mergedCv;
+                                        continue;
+
+                                    }
+                                    if (loaclCounterValues != null)
+                                    {
+                                        // delete + blob => resolve conflict
+
+                                        using (var localValuesAsCv = RawBlobToChangeVector(context, data, loaclCounterValues, existingChangeVector))
```

Comment:
You don't need this. Can do the check directly on the change vector by asking to read by id for each of the existing db ids there

Reply:
for `loaclCounterValues`, iterate over the internal values, for each non zero etag, go get the db id and then check just that.
You only care if there are all after the delete or no, nothing else